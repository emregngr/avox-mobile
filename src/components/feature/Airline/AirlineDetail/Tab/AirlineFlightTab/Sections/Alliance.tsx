import { MaterialCommunityIcons } from '@expo/vector-icons'
import React from 'react'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'
import { AirlineSectionRow } from '@/components/feature/Airline/AirlineDetail/AirlineSectionRow'

interface AllianceProps {
  alliance: string
  iconColor: string
  title: string
}

export const Alliance = ({ alliance, iconColor, title }: AllianceProps) => (
  <AirlineSectionRow title={title}>
    <View className="items-center my-5">
      <MaterialCommunityIcons color={iconColor} name="shield" size={40} />

      <ThemedText className="mt-2.5" color="text-90" type="h4">
        {alliance}
      </ThemedText>
    </View>
  </AirlineSectionRow>
)
