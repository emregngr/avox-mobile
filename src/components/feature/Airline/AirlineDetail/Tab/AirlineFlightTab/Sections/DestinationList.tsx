import React from 'react'
import { View } from 'react-native'

import { AirlineSectionRow } from '@/components/feature/Airline/AirlineDetail/AirlineSectionRow'
import { DestinationRowCard } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Cards/DestinationRowCard'

interface DestinationListProps {
  destinations: string[]
  iconColor: string
  title: string
}

export const DestinationList = ({ destinations, iconColor, title }: DestinationListProps) => (
  <AirlineSectionRow title={title}>
    <View className="flex-row flex-wrap -mx-1">
      {destinations?.map((destination, index) => (
        <DestinationRowCard destination={destination} iconColor={iconColor} key={index} />
      ))}
    </View>
  </AirlineSectionRow>
)
