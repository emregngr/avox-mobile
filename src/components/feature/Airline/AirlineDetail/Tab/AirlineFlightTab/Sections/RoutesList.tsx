import React from 'react'

import { AirlineSectionRow } from '@/components/feature/Airline/AirlineDetail/AirlineSectionRow'
import { RouteRowCard } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Cards/RouteRowCard'
import type { AirlineRoute } from '@/types/feature/airline'

interface RoutesListProps {
  iconColor: string
  routes: AirlineRoute[]
  title: string
}

export const RoutesList = ({ iconColor, routes, title }: RoutesListProps) => (
  <AirlineSectionRow title={title}>
    {routes?.map((route, index) => (
      <RouteRowCard
        destinationIata={route.destinationIata}
        iconColor={iconColor}
        key={index}
        origin={route.origin}
      />
    ))}
  </AirlineSectionRow>
)
