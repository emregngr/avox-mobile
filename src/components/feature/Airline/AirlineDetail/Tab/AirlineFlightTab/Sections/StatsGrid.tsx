import React from 'react'
import { View } from 'react-native'

import { StatsCard } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Cards/StatsCard'

interface StatsGridProps {
  iconColor: string
  localeStrings: {
    country: string
    domesticDestinations: string
    internationalDestination: string
    totalDestination: string
  }
  stats: {
    destinationCount: number
    destinationCountries: number
    domesticConnections: number
    internationalConnections: number
  }
}

export const StatsGrid = ({ iconColor, localeStrings, stats }: StatsGridProps) => (
  <>
    <View className="flex-row gap-x-4">
      <StatsCard
        iconColor={iconColor}
        iconName="flag"
        label={localeStrings.totalDestination}
        value={stats.destinationCount}
      />
      <StatsCard
        iconColor={iconColor}
        iconName="earth"
        label={localeStrings.country}
        value={stats.destinationCountries}
      />
    </View>

    <View className="flex-row gap-x-4">
      <StatsCard
        iconColor={iconColor}
        iconName="home"
        label={localeStrings.domesticDestinations}
        value={stats.domesticConnections}
      />
      <StatsCard
        iconColor={iconColor}
        iconName="globe"
        label={localeStrings.internationalDestination}
        value={stats.internationalConnections}
      />
    </View>
  </>
)
