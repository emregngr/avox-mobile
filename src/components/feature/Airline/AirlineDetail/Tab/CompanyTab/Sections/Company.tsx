import React, { useMemo } from 'react'

import { AirlineRowItem } from '@/components/feature/Airline/AirlineDetail/AirlineRowItem'
import { AirlineSectionRow } from '@/components/feature/Airline/AirlineDetail/AirlineSectionRow'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { Airline } from '@/types/feature/airline'
import { formatNumber } from '@/utils/feature/formatNumber'

interface CompanyProps {
  airlineData: Airline
}

export const Company = ({ airlineData }: CompanyProps) => {
  const { selectedLocale } = useLocaleStore()

  const { companyInfo, isoCountry, isoRegion, operations } = airlineData || {}
  const { employeeCount, foundingYear, parentCompany, passengerCapacity } = companyInfo || {}
  const { businessModel, businessType, skytraxRating, slogan } = operations || {}

  const localeStrings = useMemo(
    () => ({
      airlineInfo: getLocale('airlineInfo'),
      businessModel: getLocale('businessModel'),
      businessType: getLocale('businessType'),
      isoCountryCode: getLocale('isoCountryCode'),
      isoRegionCode: getLocale('isoRegionCode'),
      millionPerYear: getLocale('million/year'),
      numberOfEmployees: getLocale('numberOfEmployees'),
      numberOfPassengers: getLocale('numberOfPassengers'),
      parentCompany: getLocale('parentCompany'),
      skytraxRating: getLocale('skytraxRating'),
      slogan: getLocale('slogan'),
      year: getLocale('year'),
      yearOfEstablishment: getLocale('yearOfEstablishment'),
    }),
    [selectedLocale],
  )

  return (
    <AirlineSectionRow title={localeStrings.airlineInfo}>
      <AirlineRowItem
        icon="calendar"
        label={localeStrings.yearOfEstablishment}
        value={`${foundingYear} (${new Date().getFullYear() - parseInt(foundingYear)} ${localeStrings.year})`}
      />

      <AirlineRowItem icon="business" label={localeStrings.parentCompany} value={parentCompany} />

      <AirlineRowItem
        icon="people"
        label={localeStrings.numberOfPassengers}
        value={`${passengerCapacity} ${localeStrings.millionPerYear}`}
      />

      <AirlineRowItem
        icon="people"
        label={localeStrings.numberOfEmployees}
        value={formatNumber(employeeCount)}
      />

      <AirlineRowItem
        className="capitalize"
        icon="briefcase"
        label={localeStrings.businessModel}
        value={businessModel?.replace('_', ' ')}
      />

      <AirlineRowItem
        className="capitalize"
        icon="airplane"
        label={localeStrings.businessType}
        value={businessType?.replace('_', ' ')}
      />

      <AirlineRowItem icon="flag" label={localeStrings.isoCountryCode} value={isoCountry} />

      <AirlineRowItem icon="location" label={localeStrings.isoRegionCode} value={isoRegion} />

      <AirlineRowItem
        icon="star"
        label={localeStrings.skytraxRating}
        value={`${skytraxRating?.toFixed(1)} / 5`}
      />

      <AirlineRowItem icon="chatbox-ellipses" label={localeStrings.slogan} value={slogan} />
    </AirlineSectionRow>
  )
}
