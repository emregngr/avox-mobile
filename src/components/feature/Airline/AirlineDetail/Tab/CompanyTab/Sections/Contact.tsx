import { router } from 'expo-router'
import React, { useCallback, useMemo } from 'react'
import { Linking } from 'react-native'

import { AirportRowItem, AirportSectionRow } from '@/components/feature/Airport'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { Airline } from '@/types/feature/airline'

interface ContactProps {
  companyInfo: Airline['companyInfo']
  operations: Airline['operations']
}

export const Contact = ({ companyInfo, operations }: ContactProps) => {
  const { selectedLocale } = useLocaleStore()

  const { contactInfo, website } = companyInfo || {}
  const { region } = operations || {}
  const { email, phone } = contactInfo || {}

  const handlePhonePress = useCallback(async () => {
    if (phone) {
      await Linking?.openURL(`tel:${phone}`)
    }
  }, [phone])

  const handleEmailPress = useCallback(async () => {
    if (email) {
      await Linking?.openURL(`mailto:${email}`)
    }
  }, [email])

  const handleWebsitePress = useCallback(() => {
    if (website) {
      const formattedUrl = website.startsWith('http') ? website : `https://${website}`
      const regionLower = region?.toLowerCase()

      router.navigate({
        params: {
          regionLower,
          selectedWebsite: website,
          webViewUrl: formattedUrl,
        },
        pathname: '/web-view-modal',
      })
    }
  }, [website, region])

  const localeStrings = useMemo(
    () => ({
      contactInformation: getLocale('contactInformation'),
      email: getLocale('email'),
      phone: getLocale('phone'),
      website: getLocale('website'),
    }),
    [selectedLocale],
  )

  return (
    <AirportSectionRow title={localeStrings.contactInformation}>
      <AirportRowItem
        icon="globe"
        label={localeStrings.website}
        onPress={handleWebsitePress}
        value={website}
      />

      <AirportRowItem
        icon="call"
        label={localeStrings.phone}
        onPress={handlePhonePress}
        value={phone}
      />

      <AirportRowItem
        icon="mail"
        label={localeStrings.email}
        onPress={handleEmailPress}
        value={email}
      />
    </AirportSectionRow>
  )
}
