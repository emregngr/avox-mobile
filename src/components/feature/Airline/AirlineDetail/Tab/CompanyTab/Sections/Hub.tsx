import React, { useMemo } from 'react'

import { AirlineRowItem } from '@/components/feature/Airline/AirlineDetail/AirlineRowItem'
import { AirlineSectionRow } from '@/components/feature/Airline/AirlineDetail/AirlineSectionRow'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { AirlineType } from '@/types/feature/airline'

interface HubProps {
  operations: AirlineType['operations']
}

export const Hub = ({ operations }: HubProps) => {
  const { selectedLocale } = useLocaleStore()

  const { country, hub } = operations ?? {}
  const { city } = hub ?? {}

  const localeStrings = useMemo(
    () => ({
      city: getLocale('city'),
      country: getLocale('country'),
      hubInformation: getLocale('hubInformation'),
    }),
    [selectedLocale],
  )

  return (
    <AirlineSectionRow title={localeStrings.hubInformation}>
      <AirlineRowItem icon="city-variant-outline" label={localeStrings.city} value={city} />

      <AirlineRowItem icon="earth" label={localeStrings.country} value={country} />
    </AirlineSectionRow>
  )
}
