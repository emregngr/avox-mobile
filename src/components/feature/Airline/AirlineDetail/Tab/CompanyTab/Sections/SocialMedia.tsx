import * as Linking from 'expo-linking'
import React, { useCallback, useMemo } from 'react'
import { TouchableOpacity, View } from 'react-native'

import InstagramIcon from '@/assets/icons/instagram.svg'
import LinkedinIcon from '@/assets/icons/linkedin.svg'
import TiktokIcon from '@/assets/icons/tiktok.svg'
import XIcon from '@/assets/icons/x.svg'
import { AirlineSectionRow } from '@/components/feature/Airline/AirlineDetail/AirlineSectionRow'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { AirlineType } from '@/types/feature/airline'

interface SocialMediaProps {
  companyInfo: AirlineType['companyInfo']
}

export const SocialMedia = ({ companyInfo }: SocialMediaProps) => {
  const { selectedLocale } = useLocaleStore()
  const { selectedTheme } = useThemeStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { socialMedia: { instagram, linkedin, tiktok, x } = {} } = companyInfo ?? {}

  const handleSocialMediaPress = useCallback(async (url: string) => {
    await Linking?.openURL(url)
  }, [])

  const localeStrings = useMemo(
    () => ({
      socialMedia: getLocale('socialMedia'),
    }),
    [selectedLocale],
  )

  const socialMediaIcons = useMemo(
    () => (
      <View className="flex-row items-center justify-evenly mt-2">
        {instagram ? (
          <TouchableOpacity
            activeOpacity={0.7}
            hitSlop={10}
            onPress={() => handleSocialMediaPress(instagram)}
            testID="Instagram"
          >
            <InstagramIcon color={colors?.onPrimary100} height={36} width={36} />
          </TouchableOpacity>
        ) : null}

        {tiktok ? (
          <TouchableOpacity
            activeOpacity={0.7}
            hitSlop={10}
            onPress={() => handleSocialMediaPress(tiktok)}
            testID="Tiktok"
          >
            <TiktokIcon color={colors?.onPrimary100} height={36} width={36} />
          </TouchableOpacity>
        ) : null}

        {x ? (
          <TouchableOpacity
            activeOpacity={0.7}
            hitSlop={10}
            onPress={() => handleSocialMediaPress(x)}
            testID="X"
          >
            <XIcon color={colors?.onPrimary100} height={36} width={36} />
          </TouchableOpacity>
        ) : null}

        {linkedin ? (
          <TouchableOpacity
            activeOpacity={0.7}
            hitSlop={10}
            onPress={() => handleSocialMediaPress(linkedin)}
            testID="Linkedin"
          >
            <LinkedinIcon color={colors?.onPrimary100} height={36} width={36} />
          </TouchableOpacity>
        ) : null}
      </View>
    ),
    [colors?.onPrimary100, handleSocialMediaPress, instagram, tiktok, x, linkedin],
  )

  return <AirlineSectionRow title={localeStrings.socialMedia}>{socialMediaIcons}</AirlineSectionRow>
}
