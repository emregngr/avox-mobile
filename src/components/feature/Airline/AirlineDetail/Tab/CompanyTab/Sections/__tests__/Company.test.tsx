import { render } from '@testing-library/react-native'
import React from 'react'

import { Company } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Company'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import { formatNumber } from '@/utils/feature/formatNumber'

jest.mock('@/locales/i18next')

const mockedGetLocale = getLocale as jest.MockedFunction<typeof getLocale>

jest.mock('@/store/locale')

const mockedUseLocaleStore = useLocaleStore as jest.MockedFunction<typeof useLocaleStore>

jest.mock('@/utils/feature/formatNumber')

const mockedFormatNumber = formatNumber as jest.MockedFunction<typeof formatNumber>

const mockedAirlineSectionRow = jest.fn()
const mockedAirlineRowItem = jest.fn()

jest.mock('@/components/feature/Airline/AirlineDetail/AirlineSectionRow', () => {
  const { View } = require('react-native')
  return {
    AirlineSectionRow: (props: any) => {
      mockedAirlineSectionRow(props)
      return (
        <View testID="mocked-section-row" {...props}>
          {props.children}
        </View>
      )
    },
  }
})

jest.mock('@/components/feature/Airline/AirlineDetail/AirlineRowItem', () => {
  const { Text } = require('react-native')
  return {
    AirlineRowItem: (props: any) => {
      mockedAirlineRowItem(props)
      return (
        <Text testID={`row-item-${props.icon}`}>
          {props.label}: {props.value}
        </Text>
      )
    },
  }
})

const mockedAirlineData: any = {
  id: '1',
  name: 'Turkish Airlines',
  icao: 'THY',
  iata: 'TK',
  isoCountry: 'TR',
  isoRegion: 'EU',
  companyInfo: {
    employeeCount: 45000,
    foundingYear: '1933',
    parentCompany: 'Turkish Airlines Group',
    passengerCapacity: 80.2,
  },
  operations: {
    businessModel: 'passenger',
    businessType: 'major_international',
    skytraxRating: 4.2,
    slogan: 'Widen Your World',
  },
}

beforeEach(() => {
  mockedUseLocaleStore.mockReturnValue({ selectedLocale: 'en' })

  mockedGetLocale.mockImplementation((key: string) => {
    const translations: Record<string, string> = {
      airlineInfo: 'Airline Information',
      businessModel: 'Business Model',
      businessType: 'Business Type',
      isoCountryCode: 'ISO Country Code',
      isoRegionCode: 'ISO Region Code',
      millionPerYear: 'million/year',
      numberOfEmployees: 'Number of Employees',
      numberOfPassengers: 'Number of Passengers',
      parentCompany: 'Parent Company',
      skytraxRating: 'Skytrax Rating',
      slogan: 'Slogan',
      year: 'years',
      yearOfEstablishment: 'Year of Establishment',
      cargo: 'Cargo',
      passenger: 'Passenger',
      lowCost: 'Low Cost',
      regional: 'Regional',
      majorInternational: 'Major International',
      charterAirline: 'charter airline',
    }
    return translations[key] || key
  })

  mockedFormatNumber.mockImplementation(num => num?.toLocaleString() || '0')
})

describe('Company Component', () => {
  it('should render AirlineSectionRow with correct title', () => {
    render(<Company airlineData={mockedAirlineData} />)

    expect(mockedAirlineSectionRow).toHaveBeenCalledWith(
      expect.objectContaining({
        title: 'Airline Information',
      }),
    )
  })

  it('should render all AirlineRowItem components with correct props', () => {
    render(<Company airlineData={mockedAirlineData} />)

    expect(mockedAirlineRowItem).toHaveBeenCalledTimes(10)

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'calendar-month-outline',
        label: 'Year of Establishment',
        value: `1933 (${new Date().getFullYear() - 1933} years)`,
      }),
    )

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'domain',
        label: 'Parent Company',
        value: 'Turkish Airlines Group',
      }),
    )

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'account-group-outline',
        label: 'Number of Passengers',
        value: '80.2 million/year',
      }),
    )

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'account-group-outline',
        label: 'Number of Employees',
        value: '45,000',
      }),
    )

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'briefcase-outline',
        label: 'Business Model',
        value: 'Passenger',
      }),
    )

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'airplane',
        label: 'Business Type',
        value: 'Major International',
      }),
    )

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'flag-outline',
        label: 'ISO Country Code',
        value: 'TR',
      }),
    )

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'map-marker-outline',
        label: 'ISO Region Code',
        value: 'EU',
      }),
    )

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'star-outline',
        label: 'Skytrax Rating',
        value: '4.2 / 5',
      }),
    )

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'chat-outline',
        label: 'Slogan',
        value: 'Widen Your World',
      }),
    )
  })

  it('should handle cargo business model correctly', () => {
    const cargoAirlineData = {
      ...mockedAirlineData,
      operations: {
        ...mockedAirlineData.operations,
        businessModel: 'cargo',
      },
    }

    render(<Company airlineData={cargoAirlineData} />)

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'briefcase-outline',
        label: 'Business Model',
        value: 'Cargo',
      }),
    )
  })

  it('should handle low_cost business type correctly', () => {
    const lowCostAirlineData = {
      ...mockedAirlineData,
      operations: {
        ...mockedAirlineData.operations,
        businessType: 'low_cost',
      },
    }

    render(<Company airlineData={lowCostAirlineData} />)

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'airplane',
        label: 'Business Type',
        value: 'Low Cost',
      }),
    )
  })

  it('should handle regional business type correctly', () => {
    const regionalAirlineData = {
      ...mockedAirlineData,
      operations: {
        ...mockedAirlineData.operations,
        businessType: 'regional',
      },
    }

    render(<Company airlineData={regionalAirlineData} />)

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'airplane',
        label: 'Business Type',
        value: 'Regional',
      }),
    )
  })

  it('should handle unknown business model by replacing underscores', () => {
    const unknownBusinessModelData = {
      ...mockedAirlineData,
      operations: {
        ...mockedAirlineData.operations,
        businessModel: 'mixed_cargo passenger',
      },
    }

    render(<Company airlineData={unknownBusinessModelData} />)

    const businessModelCalls = mockedAirlineRowItem.mock.calls.filter(
      call => call[0].icon === 'briefcase-outline' && call[0].label === 'Business Model',
    )

    expect(businessModelCalls).toHaveLength(1)
    expect(businessModelCalls[0][0]).toEqual(
      expect.objectContaining({
        icon: 'briefcase-outline',
        label: 'Business Model',
        value: 'mixed cargo passenger',
      }),
    )
  })

  it('should handle unknown business type by replacing underscores', () => {
    const unknownBusinessTypeData = {
      ...mockedAirlineData,
      operations: {
        ...mockedAirlineData.operations,
        businessType: 'charter_airline',
      },
    }

    render(<Company airlineData={unknownBusinessTypeData} />)

    const businessTypeCalls = mockedAirlineRowItem.mock.calls.filter(
      call => call[0].icon === 'airplane' && call[0].label === 'Business Type',
    )

    expect(businessTypeCalls).toHaveLength(1)
    expect(businessTypeCalls[0][0]).toEqual(
      expect.objectContaining({
        icon: 'airplane',
        label: 'Business Type',
        value: 'charter airline',
      }),
    )
  })

  it('should handle undefined airlineData gracefully', () => {
    render(<Company airlineData={undefined as any} />)

    expect(mockedAirlineSectionRow).toHaveBeenCalledWith(
      expect.objectContaining({
        title: 'Airline Information',
      }),
    )

    expect(mockedAirlineRowItem).toHaveBeenCalledTimes(10)
  })

  it('should handle partial companyInfo data', () => {
    const partialData: any = {
      ...mockedAirlineData,
      companyInfo: {
        foundingYear: '1950',
      },
    }

    render(<Company airlineData={partialData} />)

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'calendar-month-outline',
        label: 'Year of Establishment',
        value: `1950 (${new Date().getFullYear() - 1950} years)`,
      }),
    )

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'domain',
        label: 'Parent Company',
        value: undefined,
      }),
    )
  })

  it('should handle partial operations data', () => {
    const partialData: any = {
      ...mockedAirlineData,
      operations: {
        skytraxRating: 3.8,
      },
    }

    render(<Company airlineData={partialData} />)

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'star-outline',
        label: 'Skytrax Rating',
        value: '3.8 / 5',
      }),
    )

    expect(mockedAirlineRowItem).toHaveBeenCalledWith(
      expect.objectContaining({
        icon: 'chat-outline',
        label: 'Slogan',
        value: undefined,
      }),
    )
  })

  it('should call formatNumber for employee count', () => {
    render(<Company airlineData={mockedAirlineData} />)

    expect(mockedFormatNumber).toHaveBeenCalledWith(45000)
  })

  it('should call getLocale for all required translations', () => {
    render(<Company airlineData={mockedAirlineData} />)

    const calledKeys = mockedGetLocale.mock.calls.map(call => call[0])
    const uniqueCalledKeys = [...new Set(calledKeys)]

    const requiredKeys = [
      'airlineInfo',
      'businessModel',
      'businessType',
      'isoCountryCode',
      'isoRegionCode',
      'numberOfEmployees',
      'numberOfPassengers',
      'parentCompany',
      'skytraxRating',
      'slogan',
      'yearOfEstablishment',
      'passenger',
      'majorInternational',
    ]

    requiredKeys.forEach(key => {
      expect(uniqueCalledKeys).toContain(key)
    })

    expect(uniqueCalledKeys.includes('year') || uniqueCalledKeys.includes('years')).toBe(true)
  })

  it('should memoize locale strings based on selectedLocale', () => {
    const { rerender } = render(<Company airlineData={mockedAirlineData} />)

    const initialCallCount = mockedGetLocale.mock.calls.length

    rerender(<Company airlineData={mockedAirlineData} />)

    expect(mockedGetLocale.mock.calls.length).toBe(initialCallCount)
  })

  it('should handle zero skytrax rating', () => {
    const zeroRatingData = {
      ...mockedAirlineData,
      operations: {
        ...mockedAirlineData.operations,
        skytraxRating: 0,
      },
    }

    render(<Company airlineData={zeroRatingData} />)

    const skytraxRatingCalls = mockedAirlineRowItem.mock.calls.filter(
      call => call[0].icon === 'star-outline' && call[0].label === 'Skytrax Rating',
    )

    expect(skytraxRatingCalls).toHaveLength(1)
    expect(skytraxRatingCalls[0][0]).toEqual(
      expect.objectContaining({
        icon: 'star-outline',
        label: 'Skytrax Rating',
        value: '0.0 / 5',
      }),
    )
  })
})

describe('Company Component Snapshot', () => {
  it('should render the Company Component successfully', () => {
    const { toJSON } = render(<Company airlineData={mockedAirlineData} />)

    expect(toJSON()).toMatchSnapshot()
  })
})
