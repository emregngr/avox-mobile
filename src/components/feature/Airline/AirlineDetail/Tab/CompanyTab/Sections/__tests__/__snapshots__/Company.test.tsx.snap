// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Company Component Snapshot should render the Company Component successfully 1`] = `
<View
  testID="mocked-section-row"
  title="Airline Information"
>
  <Text
    testID="row-item-calendar-month-outline"
  >
    Year of Establishment
    : 
    1933 (92 years)
  </Text>
  <Text
    testID="row-item-domain"
  >
    Parent Company
    : 
    Turkish Airlines Group
  </Text>
  <Text
    testID="row-item-account-group-outline"
  >
    Number of Passengers
    : 
    80.2 million/year
  </Text>
  <Text
    testID="row-item-account-group-outline"
  >
    Number of Employees
    : 
    45,000
  </Text>
  <Text
    testID="row-item-briefcase-outline"
  >
    Business Model
    : 
    Passenger
  </Text>
  <Text
    testID="row-item-airplane"
  >
    Business Type
    : 
    Major International
  </Text>
  <Text
    testID="row-item-flag-outline"
  >
    ISO Country Code
    : 
    TR
  </Text>
  <Text
    testID="row-item-map-marker-outline"
  >
    ISO Region Code
    : 
    EU
  </Text>
  <Text
    testID="row-item-star-outline"
  >
    Skytrax Rating
    : 
    4.2 / 5
  </Text>
  <Text
    testID="row-item-chat-outline"
  >
    Slogan
    : 
    Widen Your World
  </Text>
</View>
`;
